core/consensus/grandpa/impl/authority_manager_impl.cpp

```diff
@@ -1105,23 +921,38 @@ namespace kagome::consensus::grandpa {
   }
 
   std::shared_ptr<ScheduleNode> AuthorityManagerImpl::getNode(
-      const primitives::BlockInfo &block) const {
+      const primitives::BlockContext &context) const {
     BOOST_ASSERT(root_ != nullptr);
 
+    // Caching getter of direct chain best block
+    auto get_block =
+        [&, block = std::optional<primitives::BlockInfo>()]() mutable {
+          if (not block.has_value()) {
+            if (context.header.has_value()) {
+              const auto &header = context.header.value().get();
+              block.emplace(header.number - 1, header.parent_hash);
+            } else {
+              block.emplace(context.block);
```

Comment:
Under what circumstances does this occur?

Reply:
When block is not in a tree, we should use parent block to check direct chain existing. In other case (block is already in tree) we can use block info of block for that (ex. while digests reapplying at start).

Reply:
So when context.header has value that means the block is not in the tree?

Reply:
No, it doesn't.